src/Serilog/Capturing/PropertyValueConverter.cs

```diff
@@ -169,6 +185,61 @@ LogEventPropertyValue CreatePropertyValue(object? value, Destructuring destructu
         return new ScalarValue(value.ToString() ?? "");
     }
 
+    Destructuring? GetCachedFallbackDestructuring(
+#if !NET5_0
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+#endif
+        Type type)
+    {
+        if (_fallbackDestructuringCache.TryGetValue(type, out var fallbackDestructuring)) return fallbackDestructuring;
+
+        var fallback = GetFallbackDestructuring(type);
+        _fallbackDestructuringCache[type] = fallback;
+
+        return fallback;
+    }
+
+    Destructuring? GetFallbackDestructuring(
+#if !NET5_0
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)]
+#endif
+        Type type)
+    {
+        // Check direct hit:
```

Comment:
This is the quick-exit and COULD be merged with the base-type checks, but I preferred to have it seperately and before checking the interfaces as it would be the best-fit and most common case.